Use base images, fix travis tests, build under non-root user, add ubuntu bionic image.

lyndonhenry opened the pull request:
- Separate images into base and non-base; base images do not have souffle, non-base images do and depend on base images.
- Add image for Ubuntu Bionic, includes MPI packages in anticipation of MPI mode for Souffle (coming soon, hopefully).
- Change Travis tests to build Souffle only, not run the full test suite as this is more appropriately done in the main Souffle repo.
- Add a "souffle" user to the images, and build Souffle under that user at /home/souffle/souffle as doing everything as root causes problems for MPI.

mmcgr commented:
The idea behind doing `make check` was to test that souffle has all the dependencies it needs to run. `make check` is overkill, since running each option with a dependency once is enough, but `make` alone will not test that it runs. `SOUFFLE_CATEGORY=FastEvaluation,Provenance make check` does a much faster test, with provenance there as it's more C++11 dependent.

Note that even `make check` does not check that souffle was installed correctly., and there's no `make checkinstall` implemented. In the main repo we have this basic test:

```
make install

#generate a simple program
A=".decl A()
A().
.printsize A"

echo "$A" > a.dl

# test if an installed souffle compiles and runs
souffle -c a.dl
```

lyndonhenry commented:
Changed things to use `SOUFFLE_CATEGORY=FastEvaluation` and run `make check` in `.travis.yml`. Also added `SOUFFLE_CONFIGURE_OPTIONS="--disable-sqlite --disable-libz --disable-provenance"` as I'm not sure which of these options work in each of the images, and don't have an opportunity to check it out right now. Ideally, nothing should be disabled within the Makefile, and individual options disabled within individual containers where they do not work (including for options which must be explicitly enabled, to ensure that they cannot be).

lyndonhenry commented:
Closing this as per PR #764 over at souffle-lang/souffle. Upstream lyndonhenry/souffle-docker has also been deleted. Could someone with the power to do so please delete this repository? It is now entirely redundant.
